### DISCUSSION TITLE
How can I uninstall audacity from source?

### DISCUSSION THREAD

**Discussion by jasonho1308** [audacity/audacity#1330] [Q&A] [answered]
I have built audacity from source, but I can't find a way to uninstall it. (I want to install it officially from snap)

Version: Audacity 3.0.4-alpha-20210713
OS: Linux
I have tried to search online for solution, they all mention about the file `install_manifest.txt`, but I cannot find it, any solution?

**Comment by SteveDaulton** [comment]
Which version of  Audacity  and which operating system?

**Reply by jasonho1308** [reply]
Audacity 3.0.4-alpha-20210713
Linux

**Comment by SteveDaulton** [comment]
I've not tested, but I'd try running `sudo make  uninstall` from inside the build  folder.

**Reply by jasonho1308** [reply]
I need to build again and run that command? (I deleted the source file before)

**Comment by SteveDaulton** [comment, marked as answer]
Since you've deleted the source files:
1. configure and build Audacity as before but do not install
2. Instead of `sudo make install` run `make --dry-run install`
3. Make a note of what would be  copied to your system (if it wasn't a dry-run)
4. Manually delete those components that are safe  to delete (be careful about deleting libraries if they could be used by other apps)

**Reply by jasonho1308** [reply]
there is over 168 lines, really need to delete it manually?

**Reply by SteveDaulton** [reply]
It depends on how thorough you want to be.
Normally, Linux distributions provide packages so that your package manager can cleanly install / uninstall without breaking anything. When installing from source code or PPAs it's a bit more risky as you may not know if any of the components installed are already used by other apps.

You can easily remove most of Audacity by simply deleting:
```

usr/local/bin/audacity
usr/local/share/audacity/
usr/local/share/appdata/audacity.appdata.xml
/usr/local/share/applications/Audacity

```
and if they exist:
```

~/.audacity
~/.audacity-data
~/.audacity-files

```

or you could try `sudo make uninstall` as suggested previously.

-----

FWIW, I don't usually install any version of Audacity other than the one  that I wish to retain for production use (usually the distribution's official package). For testing newer versions, I run them from the build target folder, with a "Portable Settings" folder to keep it's settings separate from my installed version.
![portable](https://user-images.githubusercontent.com/11681097/126318122-787a62a6-b8da-4840-8daa-9ccd4889df53.png)
The "audacity.sh" file is just a little bash script so that I can launch Audacity by double clicking  on the  script. It contains:
```

#!/usr/bin/env bash
exec ./audacity

```
and of course it must have "execute" permission enabled.

**Reply by jasonho1308** [reply]
I don't have this folder `usr/local/share/appdata/`
